File: Bin/PLSamples/Data/Scripts/Functors.py
#[-------------------------------------------------------]
#[ Global variables                                      ]
#[-------------------------------------------------------]
g_Factor = 1


#[-------------------------------------------------------]
#[ Global functions                                      ]
#[-------------------------------------------------------]
def ScriptFunction(a):
	return a

def CallCpp(a):
	return FirstNamespace['SecondNamespace']['CppFunction'](a) + CppMethod(a) + CppScriptFunction(a)

def GetFactor():
	return g_Factor

def SetFactor(a):
	global g_Factor	# If this isn't done, g_Factor is used as local variable when assigning a new value to it
	g_Factor = a

def Calculate(a, b):
	return a * b * g_Factor

def ReturnMyString(s):
	return CppStringFunction(s)

File: Bin/PLSamples/Data/Scripts/Functors.lua
--[-------------------------------------------------------]
--[ Global variables                                      ]
--[-------------------------------------------------------]
g_Factor = 1


--[-------------------------------------------------------]
--[ Global functions                                      ]
--[-------------------------------------------------------]
function ScriptFunction(a)
	return a
end

function CallCpp(a)
	-- "FirstNamespace.SecondNamespace.CppFunction(a)" is an equivalent for "FirstNamespace["SecondNamespace"]["CppFunction"](a)"
	return FirstNamespace.SecondNamespace.CppFunction(a) + CppMethod(a) + CppScriptFunction(a)
end

function GetFactor()
	return g_Factor
end

function SetFactor(a)
	g_Factor = a
end

function Calculate(a, b)
	return a * b * g_Factor
end

function ReturnMyString(s)
	return CppStringFunction(s)
end
